File: GALVOS.py
import sys
import redpitaya_scpi as scpi

IP = '10.68.9.146'
print("Connexion à la Red Pitaya sur " + IP + "...")
rp = scpi.scpi(IP)

# --- TA POSITION FIXE ---
# Modifie ces valeurs (entre -1.0 et 1.0) pour pointer où tu veux
voltage_x = 0.5
voltage_y = 0.2

print("Envoi de la commande de positionnement...")

# Les sorties de la Red Pitaya gardent la dernière tension demandée en mémoire
rp.tx_txt('SOUR1:VOLT ' + str(voltage_x))
rp.tx_txt('SOUR2:VOLT ' + str(voltage_y))

print("Miroirs stabilisés à X=" + str(voltage_x) + "V, Y=" + str(voltage_y) + "V")

# Le programme attend ici indéfiniment. Tu peux faire tes mesures avec le spectromètre.
input("\nPrends ton temps. Appuie sur Entrée quand tu as terminé pour remettre à 0V...")

# --- NETTOYAGE ET SÉCURITÉ ---
print("Remise à 0V des galvos et fermeture de la connexion.")
rp.tx_txt('SOUR1:VOLT 0')
rp.tx_txt('SOUR2:VOLT 0')
rp.close()

File: Archives/GALVOS.m
%% Configuration de la Red Pitaya (TCP/IP)
IP = '10.68.9.146'; 
port = 5000;
RP = tcpclient(IP, port);
RP.ByteOrder = "big-endian";
configureTerminator(RP, 'CR/LF');

%% Définition de la position constante
voltage_x = 1;
voltage_y = 1;

fprintf('Envoi de la commande de positionnement...\n');

%% NOUVEAU : Envoi des commandes SCPI complètes
% 1. Définir le mode en signal continu (DC)
writeline(RP, 'SOUR1:FUNC DC');
writeline(RP, 'SOUR2:FUNC DC');

% 2. Définir la tension (En mode DC, c'est l'Offset qui définit la tension)
writeline(RP, sprintf('SOUR1:VOLT:OFFS %f', voltage_x));
writeline(RP, sprintf('SOUR2:VOLT:OFFS %f', voltage_y));

% 3. ALLUMER LES SORTIES (C'est la clé !)
writeline(RP, 'OUTPUT1:STATE ON');
writeline(RP, 'OUTPUT2:STATE ON');

fprintf('Miroirs stabilisés à X = %.2f V, Y = %.2f V\n', voltage_x, voltage_y);

%% Maintien de la position
input('Appuie sur Entrée dans cette console quand tu as terminé pour remettre à 0V...', 's');

%% Sécurité : Remise à zéro et fermeture
fprintf('Remise à 0V et extinction des sorties...\n');
writeline(RP, 'SOUR1:VOLT:OFFS 0');
writeline(RP, 'SOUR2:VOLT:OFFS 0');
% Éteindre les sorties
writeline(RP, 'OUTPUT1:STATE OFF');
writeline(RP, 'OUTPUT2:STATE OFF');

clear RP;
fprintf('Connexion fermée avec succès.\n');